### ISSUE TITLE
Command to show more without scrolling to top.

### ISSUE THREAD

**Issue by bingen13** [nolanlawson/pinafore#2144]
For people using screen readers, the . command to show more and scroll moves the focus from the toot where one was. It would be good to have a command (perhaps ,) to refresh the timeline without moving focus. Otherwise it is necessary to scroll down until one finds the place where one was.

An alternative might be having a way to remember, and go to, the previous position; or for up arrow on the top of the timeline to try to fetch more toots and scroll one up.

**Comment by nolanlawson** [comment]
So if I understand correctly, the use case is:

1. I'm reading toot C
2. I see "2 new toots available"
3. I press "."
4. I am reading in reverse chronological order, so I would like the focus to remain on C and I can press up to go to B (and then on to A) at my leisure

Makes sense to me, but may be tricky to implement. If we _don't_ scroll, then we may be focused on an out-of-view toot. So I think this would have to be a keyboard shortcut that loads without scrolling.

**Comment by bingen13** [comment]
Yeah, loading new toots without scrolling would be good.

The main problem is losing the reference of C, so that anything that makes it possible to go back to it would be fine. I can think of a few ways around.

1. Some kind of bookmark position that we could use before "." and scroll, then restore position command.
2. Load without scroll (I think my favourite solution).
3. A goto command that would take a numeric argument, so when you see 19 nnew toots and "." you could do g 19 or something.

I'm also never really quite sure when Pinafore automatically refreshes a timeline or doesn't. Sometimes I press g h, do a few things, and the focus moves due to new toots coming in, sometimes I just get the n new toots button. Would be nice if this were more deterministic or even could be configured (freeze/unfreeze).

Sorry if I'm asking a lot. I haven't really looked at the code, just been using Pinafore for a few days with a screen reader and noticed a few niggles.

**Comment by nolanlawson** [comment]
You can check the checkbox for "Disable infinite scroll" in the settings if you'd like to control when new toots appear at the top or bottom of the timeline. The default logic is based on whether you're scrolled down or not.

I think the keyboard shortcut is the simplest solution.

**Comment by nolanlawson** [comment]
Tried my hand at this, but as it turns out, it's incredibly difficult to load more items at the top while maintaining scroll position and focus. Everything jumps around.

**Comment by bingen13** [comment]
Mmm, how about implementing markers? https://docs.joinmastodon.org/methods/timelines/markers/

That way could set the marker before pressing . and for going to it, it could be a loop going down toot by toot until the focused toot is the one marked. Would that be easier?

**Comment by nolanlawson** [comment]
Yeah I think that's https://github.com/nolanlawson/pinafore/issues/1995. It would be a pretty big overhaul of the way Pinafore's current timeline/notification system works.
